roxctl/scanner/downloaddb/downloaddb.go

```diff
@@ -0,0 +1,293 @@
+package downloaddb
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/spf13/cobra"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/utils"
+	pkgVersion "github.com/stackrox/rox/pkg/version"
+	"github.com/stackrox/rox/roxctl/common/environment"
+	"github.com/stackrox/rox/roxctl/common/flags"
+)
+
+const (
+	contentLengthHdrKey = "Content-Length"
+
+	bundleFileNameFmt    = "%[1]s/scanner-vulns-%[1]s.zip"
+	latestBundleFileName = "scanner-vuln-updates.zip"
+)
+
+type scannerDownloadDBCommand struct {
+	// Properties that are bound to cobra flags.
+	filename     string
+	force        bool
+	skipCentral  bool
+	skipVariants bool
+	timeout      time.Duration
+	version      string
+
+	// filenameValidated is set to true if filename is non-empty and has
+	// already been validated, this ensures the same file isn't validated
+	// repeatedly when processing version variants.
+	filenameValidated bool
+
+	// Properties that are injected or constructed.
+	env environment.Environment
+}
+
+func (cmd *scannerDownloadDBCommand) construct(c *cobra.Command) {
+	cmd.timeout = flags.Timeout(c)
+}
+
+func (cmd *scannerDownloadDBCommand) downloadDb() error {
+	// Get the list of file names to attempt to download.
+	bundleFileNames, err := cmd.buildBundleFileNames()
+	if err != nil {
+		return err
+	}
+
+	var errs []error
+	for _, bundleFileName := range bundleFileNames {
+		// Get the name of the output file and ensures it's valid.
+		outFileName, err := cmd.buildAndValidateOutputFileName(bundleFileName)
+		if err != nil {
+			// If there was an error validating the output file, assume the file exists
```

Comment:
I think the comment is a little misleading because it's not really an assumption. I believe this only returns an error if there is a problem with os.Stat

Reply:
Will remove it.

Reply:
https://github.com/stackrox/stackrox/pull/9803/commits/b85f3b05bd969178d5518344cf80914d21e77271